### DISCUSSION TITLE
애니메이션으로 움직이는 발판을 캐릭터가 따라가게하는법2

### DISCUSSION THREAD

**Discussion by Manjok1** [naverz/zepeto-studio-kor#1128] [5. Q&A]
안녕하세요 #534 번 질문을 보고 따라하고있는데 잘 안돼서 혹시 도움주실 분 있나해서 올려봐요
전에 글을 한 번 올렸는데 답변이 달리지않아 다시 올립니다

![image](https://user-images.githubusercontent.com/88345350/182918855-106734f6-4d70-48d2-a957-46343ee2d08d.png)
이 코드를 똑같이 따라해봤는데 제가 놓치는 부분이 있는 것 같아서요.

![image](https://user-images.githubusercontent.com/88345350/182919039-164bfb76-fd96-4be4-99f9-cacda1eed1eb.png)
![image](https://user-images.githubusercontent.com/88345350/182919185-781bd633-18cc-454f-ba3a-1f959d1a8ff6.png)
(Spinning Cube에 애니메이터와 콜라이더가 붙어있고, 자식인 큐브는 메쉬 렌더러만 가진 상태)
이런식을으로 상자를 딛을 수 있는 콜라이더와
is trigger을 체크한 캐릭터가 들어간걸 감지할 콜라이더를 만들었고

코드의 "Moving Box"에 제가 만든 Spinning Cube자체를 넣는건 알겠는데
detect에는 무엇을 넣어야할지 잘 모르겠습니다.

OnTriggerExit의 if문을 생략하고 바로 Setparent(null)을 해주니 캐릭터가 이상하게 튀는 현상이 발생해서...
detect가 뭔지 알아야할 것 같네요. 혹시 도움주실 분 있으면 정말 감사할 것 같습니다. 

만들어보고 싶은 월드에 꼭 필요한 기능이어서 해결하고싶습니다 ㅠㅠ.

**Comment by ZdevMonZ** [comment]
무빙 블록 샘플 코드도 한번 보시면 좋을 듯요 https://github.com/naverz/world-movingblock-sample/blob/main/Assets/Scripts/MovingBlock.ts

**Reply by Manjok1** [reply]
답변감사합니다 그런데 애니메이터를 이용해 발판 및 그외 오브젝트들을 움직일 것이기때문에 지금 방법을 고민하고있는거여서요. 샘플을 활용하긴 어려울 것 같아요.

**Comment by IndiaInk10** [comment]
별도의 Position이나 Rotation 세팅주지마시고, 그냥 SetParent만 해줘보세요.
![image](https://user-images.githubusercontent.com/49700162/183051535-dba245f4-39ed-4a8f-981a-2c2346cb8ec1.png)
